File: SantanderTeste/app/src/main/java/com/example/alessandrofsouza/santanderapp/presentation/pages/investment/InvestmentContract.java
package com.example.alessandrofsouza.santanderapp.presentation.pages.investment;

import com.example.alessandrofsouza.santanderapp.domain.model.Screen;
import com.example.alessandrofsouza.santanderapp.presentation.pages.base.BasePresenter;
import com.example.alessandrofsouza.santanderapp.presentation.pages.base.BaseView;

public interface InvestmentContract {

    interface View extends BaseView<Presenter> {
        void showInvestmentScreen(Screen screen);
    }

    interface Presenter extends BasePresenter {
        void getInvestmentScreen();
    }

}

File: SantanderInvestimento/app/src/main/java/br/com/santander/santanderinvestimento/investiment/presentation/InvestmentContract.kt
package br.com.santander.santanderinvestimento.investiment.presentation

import br.com.santander.santanderinvestimento.core.presentation.BasePresenter
import br.com.santander.santanderinvestimento.core.presentation.BaseView
import br.com.santander.santanderinvestimento.investiment.domain.entity.Investment

interface InvestmentContract {

    interface View : BaseView<Presenter> {
        fun showSuccess(investment: Investment)
    }


    interface Presenter : BasePresenter<View> {
        fun loadFeed()
    }

}